experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/editableTreeTypes.spec.ts

```diff
@@ -171,4 +180,133 @@ describe("editableTreeTypes", () => {
 		type _3 = requireFalse<isAssignableTo<typeof basicStruct, MapSchema>>;
 		type _4 = requireTrue<isAssignableTo<typeof basicStruct, StructSchema>>;
 	}
+
+	function nominalTyping(): void {
+		const builder2 = new SchemaBuilder("test");
+		const emptyStruct1 = builder2.struct("empty1", {});
+		const emptyStruct2 = builder2.struct("empty2", {});
+
+		// Schema for types which only different in name are distinguished
+		{
+			type _1 = requireFalse<isAssignableTo<typeof emptyStruct1, typeof emptyStruct2>>;
+			type _2 = requireFalse<isAssignableTo<typeof emptyStruct2, typeof emptyStruct1>>;
+		}
+		type Empty1 = TypedNode<typeof emptyStruct1>;
+		type Empty2 = TypedNode<typeof emptyStruct2>;
+
+		// Schema for TypedNode which only different in name are distinguished
+		{
+			// TODO: Fis this. Might be fixed when moving to class based schema builder. Otherwise add strongly typed named to nodes.
+			// @ts-expect-error TODO: fix this and remove expected error.
+			type _1 = requireFalse<isAssignableTo<Empty1, Empty2>>;
+			// @ts-expect-error TODO: fix this and remove expected error.
+			type _2 = requireFalse<isAssignableTo<Empty2, Empty1>>;
+		}
+	}
+
+	type Basic1 = TypedNode<typeof basicStruct>;
```

Comment:
nit: call this `BasicStruct` and call `Basic2` `BasicFieldNode` so the test cases are a little easier to read.

Reply:
Done.